scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java

```diff
@@ -61,64 +69,64 @@ private boolean isInFinishedState(InternalJob job) {
+
+                for (InternalJob job : jobs) {
+                    // if the job was already finished we just remove it from the context
+                    if (isInFinishedState(job)) {
+                        service.submitTerminationDataHandler(service.getJobs().removeJob(job.getId()));
+                    } else {
+                        service.submitTerminationDataHandler(service.getJobs().killJob(job.getId()));
+                    }
+                    job.setRemovedTime(removedTime);
                 }
-            }
-            service.submitTerminationDataHandler(terminationData);
-
-            // if the job doesn't exist in the DB anymore we can stop here
-            if (jobs.size() != 1) {
-                return false;
-            }
-
-            InternalJob job = jobs.get(0);
 
-            job.setRemovedTime(System.currentTimeMillis());
-
-            boolean removeFromDb = PASchedulerProperties.JOB_REMOVE_FROM_DB.getValueAsBoolean();
-
-            dbManager.removeJob(jobId, job.getRemovedTime(), removeFromDb);
-
-            ServerJobAndTaskLogs.getInstance().remove(jobId, job.getOwner());
-
-            if (logger.isInfoEnabled()) {
-                logger.info("Job " + jobId + " removed in " + (System.currentTimeMillis() - start) + "ms");
+                boolean removeFromDb = PASchedulerProperties.JOB_REMOVE_FROM_DB.getValueAsBoolean();
+                dbManager.removeJob(jobIds, removedTime, removeFromDb);
+
+                for (InternalJob job : jobs) {
+                    ServerJobAndTaskLogs.getInstance().remove(job.getId(), job.getOwner());
+                    if (logger.isInfoEnabled()) {
+                        logger.info("Job " + job.getId() + " removed in " + (System.currentTimeMillis() - start) +
+                                    "ms");
+                    }
+                    // send event to front-end
+                    service.getListener()
+                           .jobStateUpdated(job.getOwner(),
+                                            new NotificationData<>(SchedulerEvent.JOB_REMOVE_FINISHED,
+                                                                   new JobInfoImpl((JobInfoImpl) job.getJobInfo())));
+                }
+                allJobsWereRemoved = true;
             }
 
-            // send event to front-end
-            service.getListener()
-                   .jobStateUpdated(job.getOwner(),
-                                    new NotificationData<JobInfo>(SchedulerEvent.JOB_REMOVE_FINISHED,
-                                                                  new JobInfoImpl((JobInfoImpl) job.getJobInfo())));
-
             service.wakeUpSchedulingThread();
         } catch (Exception e) {
-            logger.error("Error while removing job " + jobId, e);
+            logger.error("Error while removing list of jobs " +
+                         jobIds.stream().map(JobId::value).collect(Collectors.joining(", ")), e);
```

Comment:
Putting e.getMessage() would ease the debugging.

Reply:
Thanks